extensions/2.0/Khronos/KHR_node_selectability/README.md

```diff
@@ -0,0 +1,107 @@
+
+The concept of “selectability” defines whether the selection ray should terminate at a given object. A selection action should be interpreted as applying to the first “selectable” object along the ray, with all earlier “unselectable” objects being skipped, regardless of the opacity, visibility or appearance of the objects.
+
+## Extending Nodes
+
+The `KHR_node_selectability` extension object is added to the objects within the `nodes` array. The extension object contains a single boolean `selectable` property. This value is mutable through JSON pointers as defined in the glTF 2.0 Asset Object Model and controls “selectability” of the node that contains it and all its children nodes recursively. A value of `false` causes all nodes below in the hierarchy to be omitted from “selection”, even any nodes below that have a value of `true`.
+
+|                | Type      | Description                               | Required            |
+|----------------|-----------|-------------------------------------------|---------------------|
+| **selectable** | `boolean` | Specifies whether the node is selectable. | No, default: `true` |
+
+In other words, a node is “selectable” if and only if there exists no node including or above it in the hierarchy where such node has a `selectable` property with value `false`. This allows a single change of a `selectable` property at a high level of the hierarchy to enable or disable the selection of complex (multi-node) objects.
+
+In the following example, both nodes (and therefore their meshes) are initially unselectable.
+
+```json
+{
+    "nodes": [
+        {
+            "children": [1],
+            "mesh": 0,
+            "extensions": {
+                "KHR_node_selectability": {
+                    "selectable": false
+                }
+            },
+        },
+        {
+            "mesh": 1
+        }
+    ]
+}
+```
+
+## Extending glTF 2.0 Asset Object Model
+
+The following pointer template represents the mutable property defined by this extension.
+
+| Pointer                                                  | Type   |
+|----------------------------------------------------------|--------|
+| `/nodes/{}/extensions/KHR_node_selectability/selectable` | `bool` |
+
+## Interaction with KHR_interactivity
+
+Unless used in conjunction with `KHR_interactivity`, the result of a “select” action on a given object is left up to the implementation.
+
+When used in conjunction with `KHR_interactivity`, a new interactivity event is defined as follows:
+
+|                         |                  | |
+|-------------------------|------------------|-|
+| **Type**                | `event/onSelect` | Select event |
+| **Configuration**       | `int nodeIndex`        | Index of a node |
+|                         | `bool stopPropagation` | Whether to allow parents of this node to also receive the event |
+| **Output values**       | `int selectedNodeIndex`  | Index of the actual node that was selected |
```

Comment:
What is the distinction between the configuration `int nodeIndex` and the output value `int selectedNodeIndex`? Wouldn't they just be set to the same value?

The description below defines `selectedNodeIndex` is initialized to -1, so is the distinction that `selectedNodeIndex` can either be -1 or set to `nodeIndex`?

Reply:
nodeIndex is in the configuration, meaning that when you put an onSelect node into your node graph, you choose a nodeIndex to listen to. selectedNodeIndex is an output, meaning that it is a value that is set at runtime when the onSelect event occurs. If you set your nodeIndex to a node that has no children, then selectedNodeIndex will always be set to nodeIndex. If you set your nodeIndex to be a node that has children, then selectedNodeIndex may be any node (that can accept select events) within the subtree of nodeIndex, depending on which specific node was selected.

Reply:
Ah, I see, that makes sense. So it allows selections to "bubble up", reaching an event listening for an ancestor, but still point to a specific child. Thanks!

Reply:
So, maybe rename `nodeIndex` to `listenerNodeIndex`?

Reply:
That's implied by the value being an _event_ configuration. Let's clarify the description instead.